pkg/scheduler/ugm/manager.go

```diff
@@ -401,6 +390,27 @@ func (m *Manager) clearEarlierSetLimits(newUserLimits map[string]map[string]*Lim
 	m.clearEarlierSetUserLimits(newUserLimits)
 }
 
+// clearEarlierSetUserWildCardLimits Traverse new wild card user config and decide whether earlier usage needs to be cleared or not
+// by comparing with the existing config. If config set earlier not now, then traverse all users, check whether wild card limit has been applied/used or not.
+// Reset earlier settings for the users only when wild card user limit has been applied.
+func (m *Manager) clearEarlierSetUserWildCardLimits(newUserWildCardLimits map[string]*LimitConfig) {
+	for queuePath := range m.userWildCardLimitsConfig {
+		hierarchy := strings.Split(queuePath, configs.DOT)
```

Comment:
This should be inside the if `newUserWildCardLimits`

Reply:
At times, config may not even have a single limit settings. So, we cannot depend on len of newUserWildCardLimits